allensdk/brain_observatory/behavior/swdb/save_flash_response_df.py

```diff
@@ -46,34 +75,50 @@ def get_flash_response_df(session, response_analysis_params):
     df.insert(loc=2, column='dff_trace_timestamps', value=trace_timestamps_list)
     return df
 
-def get_p_values_from_shuffled_spontaneous(session, flash_response_df, response_window_duration=0.5):
+def get_p_values_from_shuffled_spontaneous(session, flash_response_df, response_window_duration=0.5,number_of_shuffles=10000):
     '''
         Computes the P values for each cell/flash. The P value is the probability of observing a response of that
         magnitude in the spontaneous window. The algorithm is copied from VBA
+
+        INPUTS:
+            <session> a BehaviorOphysSession object
+            <flash_response_df> the flash_response_df for this session
+            <response_window_duration> is the duration of the response_window that was used to compute the mean_response in the flash_response_df. This is used here to extract an equivalent duration df/f trace from the spontaneous timepoint
+            <number_of_shuffles> the number of shuffles of spontaneous activity used to compute the pvalue
+    
+        OUTPUTS:
+        fdf, a copy of the flash_response_df with a new column appended 'p_value' which is the per-flash X per-cell p-value
+    
+        ASSERTS:
+            each p value is bounded by 0 and 1, and does not include any NaNs
+
     '''
     # Organize Data
     fdf = flash_response_df.copy()
     st  = session.stimulus_presentations.copy()
     included_flashes = fdf.index.get_level_values(1).unique()
     st  = st[st.index.isin(included_flashes)]
+
     # Get Sample of Spontaneous Frames
     spontaneous_frames = get_spontaneous_frames(session)
-    #  Set up Params
-    ophys_frame_rate = 31
+
+    #  Compute the number of response_window frames
+    ophys_frame_rate = 31 # Shouldn't hard code this here
```

Comment:
if not yet getting this from data, move it to a static variable so that it can be replaced easily